.github/workflows/release.yml

```diff
@@ -1,63 +1,124 @@
 name: Build and upload to PyPI
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.ref }}
+  cancel-in-progress: true
+
 # Only build on published releases
 on:
   release:
     types:
       - published
 
 jobs:
-  build:
-    name: Build sdist
-    runs-on: ubuntu-18.04
-    defaults:
-      run:
-        shell: bash -l {0}
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    outputs:
+      SDIST_NAME: ${{ steps.sdist.outputs.SDIST_NAME }}
 
     steps:
       - uses: actions/checkout@v3
-      - uses: conda-incubator/setup-miniconda@v2
         with:
-          activate-environment: test-environment
-          python-version: ${{ matrix.python-version }}
-          channels: conda-forge/label/testing,conda-forge
+          # We need the full history to generate the proper version number
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.11'
 
       - name: Install dependencies
+        run: python -m pip install build twine
+
+      - name: Build sdist
+        id: sdist
         run: |
-          PACKAGES="cython fiona matplotlib-base numpy pyproj pykdtree scipy"
-          PACKAGES="$PACKAGES owslib pep8 pillow pyshp pytest"
-          PACKAGES="$PACKAGES pytest-xdist setuptools_scm shapely"
-          conda install $PACKAGES
+          python -m build --sdist
+          # Get the name of the build sdist file for later use
+          echo "SDIST_NAME=$(ls -1 dist)" >> $GITHUB_OUTPUT
 
-      - name: Create sdist
-        run: python setup.py build_ext sdist
+      - name: Check README rendering for PyPI
+        run: twine check dist/*
 
-      - name: Save built packages as artifact
+      - name: Upload sdist result
         uses: actions/upload-artifact@v3
         with:
-          name: packages-${{ runner.os }}-${{ matrix.python-version }}
-          path: dist/
+          name: sdist
+          path: dist/*.tar.gz
           if-no-files-found: error
-          retention-days: 5
+
+  generate-wheels-matrix:
+    name: Generate wheels matrix
+    needs: build_sdist
```

Comment:
Does this really need the sdist?

Reply:
I figured that would just stop it from continuing on if even the sdist failed. But, I removed it for now, we shouldn't be going down this path too often anyways.